third_party/WebKit/Source/modules/vr/VRDisplay.cpp

```diff
@@ -316,6 +318,63 @@ VRPassThroughCamera* VRDisplay::getPassThroughCamera()
   return m_passThroughCamera;
 }
 
+VRAnchor* VRDisplay::createAnchor(WTF::Vector<float>& modelMatrix)
+{
+  if (!m_display)
+    return nullptr;
+
+  device::mojom::blink::VRAnchorPtr mojomAnchor;
+  m_display->CreateAnchor(modelMatrix, &mojomAnchor);
+  if (mojomAnchor.is_null()) {
+    return nullptr;
+  }
+  VRAnchor* anchor = new VRAnchor();
+  anchor->setAnchor(mojomAnchor);
+  m_anchors.push_back(anchor);
+
+  VLOG(0) << "JUDAX: VRDisplay::createAnchor -> Anchor created with identifier = " <<
+             anchor->identifier();
+
+  return anchor;
+}
+
+VRAnchor* VRDisplay::createAnchor(DOMFloat32Array* modelMatrix)
+{
+  if (!m_display)
+    return nullptr;
+
+  // Copy the DOMFloat32Array to a Vector<float>
+  WTF::Vector<float> modelMatrixVector;
+  size_t modelMatrixSize = modelMatrix->length();
+  modelMatrixVector.resize(modelMatrixSize);
+  for (size_t i = 0; i < modelMatrixSize; i++) {
+    modelMatrixVector[i] = modelMatrix->data()[i];
+  }
+
+  // Use the other overloaded method to create an anchor using the Vector.
+  return createAnchor(modelMatrixVector);
+}
+
+void VRDisplay::removeAnchor(VRAnchor* anchor)
+{
+  if (!m_display)
+    return;
+
+  bool removed = false;
+  for (size_t i = 0; !removed && i < m_anchors.size(); i++) {
+    // NOTE: Should we compare the pointers directly?
+    if (m_anchors[i]->identifier() == anchor->identifier()) {
+      m_anchors.remove(i);
+      removed = true;
```

Comment:
any reason to not just break here? you get rid of a bool and an extra check. Are you expecting multiple entries with matching ids?